### ISSUE TITLE
Traversing typedef struct issue

### ISSUE THREAD

**Issue by volium** [eliben/pyelftools#179]
This is **potentially** not a bug, but rather a gap in documentation, or even more likely, a lack of knowledge on my part.

I have a struct definition similar to this:

```
typedef struct node{
  uint32_t id;
  struct node *next;
} Node_t;
```

I'm able to use traverse the CUs until I find the first reference to the a variable of type ```Node_t```;  which I then start examining and deconstruct. My issue is that once I find the ```*next``` element in the struct definition, I am not able to link it back to the main ```Node_t``` definition, I can only see the parent which is reported to have type ```node```.

I am using pyelftools to walk the elf file produced by the compiler to generate RPC code callable from the a different host.

Clues, suggestions?

Thanks in advance!

**Comment by eliben** [comment]
Can you provide a minimal test object file with this stuff encoded in the DIE tree? Did you try to use the `dwarf_die_tree` example to dump the recursive structure of DIEs?

**Comment by volium** [comment]
Sorry it took me this long, I've been busy with work.

I haven't tried the `dwarf_die_tree` example yet, I might have some time this weekend, though, in the meantime I am attaching an `out` file that was produced from this code:

```
#include <stdlib.h>
#include <stdint.h>

typedef struct node{
  uint32_t id;
  struct node *next;
} Node_t;

int main(void)
{
  Node_t testNode0 = {0, NULL};
  Node_t testNode1 = {0, NULL};
  Node_t testNode2 = {0, NULL};
  
  testNode0.next = &testNode1;
  testNode1.next = &testNode2;
  testNode2.next = &testNode0;
  
  while(1){};
}
```

Thanks!

[test.zip](https://github.com/eliben/pyelftools/files/1712736/test.zip)

**Comment by volium** [comment]
I searched some more, and I found that I am responsible for tracking objects like these using their respective `DW_AT_type` and `offset`.

I stumbled upon issue https://github.com/eliben/pyelftools/issues/113 which pretty much describes the issue I wanted to solve. Furthermore, issue https://github.com/eliben/pyelftools/issues/27 gave me the insight on how to correctly determine the offset of the parent type.

I'm closing this now.
